micronota/summarize.py

```diff
@@ -7,22 +7,60 @@
 # ----------------------------------------------------------------------------
 
 
-def summarize(obj, types=('length', 'nuc_freq', 'CDS', 'ncRNA', 'rRNA', 'tRNA', 'tandem_repeat', 'terminator', 'CRISPR')):
+def _count_interval_features(obj, feature_type):
+    '''Count number of interval features
+
+    Parameters
+    ----------
+    obj : ``Sequence``
+        sequence with interval features
+    feature_type : str
+        type of features to count
+
+    Returns
+    -------
+    int
+        number of features of the type
+
+    Raises
+    ------
+    AttributeError
+        obj has no attribute 'interval_metadata'
+    '''
+    return len([x for x in obj.interval_metadata._intervals
```

Comment:
use the public `interval_metadata.query` instead of the private attr `._intervals`, which may be subject to changes in future